src/as-utils.c

```diff
@@ -603,58 +603,33 @@ as_ptr_array_to_strv (GPtrArray *array)
  * Returns: the number of replacements done, or 0 if @search is not found.
  **/
 guint
-as_gstring_replace (GString *string, const gchar *search, const gchar *replace)
+as_gstring_replace (GString *string, const gchar *find, const gchar *replace)
 {
-	gchar *tmp;
-	guint count = 0;
-	gsize search_idx = 0;
-	gsize replace_len;
-	gsize search_len;
+	gsize f_len, r_len, pos;
+	gchar *cur, *next;
+	guint n = 0;
 
 	g_return_val_if_fail (string != NULL, 0);
-	g_return_val_if_fail (search != NULL, 0);
+	g_return_val_if_fail (find != NULL, 0);
 	g_return_val_if_fail (replace != NULL, 0);
 
-	/* nothing to do */
-	if (string->len == 0)
-		return 0;
-
-	search_len = strlen (search);
-	replace_len = strlen (replace);
-
-	do {
-		tmp = g_strstr_len (string->str + search_idx, -1, search);
-		if (tmp == NULL)
-			break;
-
-		/* advance the counter in case @replace contains @search */
-		search_idx = (gsize) (tmp - string->str);
-
-		/* reallocate the string if required */
-		if (search_len > replace_len) {
-			g_string_erase (string,
-					(gssize) search_idx,
-					(gssize) (search_len - replace_len));
-			memcpy (tmp, replace, replace_len);
-		} else if (search_len < replace_len) {
-			g_string_insert_len (string,
-					     (gssize) search_idx,
-					     replace,
-					     (gssize) (replace_len - search_len));
-			/* we have to treat this specially as it could have
-			 * been reallocated when the insertion happened */
-			memcpy (string->str + search_idx, replace, replace_len);
-		} else {
-			/* just memcmp in the new string */
-			memcpy (tmp, replace, replace_len);
-		}
-		search_idx += replace_len;
-		count++;
-	} while (TRUE);
+	f_len = strlen (find);
+	r_len = strlen (replace);
+	cur = string->str;
+
+	while ((next = strstr (cur, find)) != NULL)
+	{
```

Comment:
Brace should be on the end of the previous line to match existing style.

Reply:
I've not fixed style like that (only spaces to tabs) as this is a direct copy-paste from glib.

Reply:
(added a comment to make that clear. I can update the style to match AS's if that's desired)

Reply:
Yes, please make it follow the style of the rest of the AppStream code :-)